pkgs/top-level/all-packages.nix

```
@@ -6955,6 +6955,8 @@ with pkgs;
 
   colormake = callPackage ../development/tools/build-managers/colormake { };
 
+  comfyui-custom-nodes = recurseIntoAttrs (callPackage ../by-name/co/comfyui/custom-nodes.nix { });
```

```suggestion
  comfyui-cuda = callPackage ../by-name/co/comfyui/package.nix { gpuBackend = "cuda"; }

  comfyui-rocm = callPackage ../by-name/co/comfyui/package.nix { gpuBackend = "rocm"; }

  comfyui-custom-nodes = recurseIntoAttrs (callPackage ../by-name/co/comfyui/custom-nodes.nix { });
```

I suggest adding the variants to make sure hydra caches them. (In ``comfyui-cuda``'s case, so we don't have to override comfyui for packages that want a cuda variant.)
We don't have to do this with the unadulterated ``comfyui``, since the ``by-name`` implementation takes care of it.

---

Note, we can also do this.
```suggestion
  comfyui = callPackage ../by-name/co/comfyui/package.nix { }

  comfyui-cuda = comfyui.override { gpuBackend = "cuda"; }

  comfyui-rocm = comfyui.override { gpuBackend = "rocm"; }

  comfyui-custom-nodes = recurseIntoAttrs (callPackage ../by-name/co/comfyui/custom-nodes.nix { });
```

---

@Madouura looks much better, but shouldn't we use the naming scheme `comfyWithRocm` to match `torchWithRocm`?

---

``withRocm`` and ``withCuda`` is arguably outdated.
We should probably eventually change it to ``torch-rocm`` (or ``torch-hip``) and ``torch-cuda``.

---

I think the correct way to.handle this is to introduce pkgsRocm/pkgsCuda similar to the pkgsStatic. The we just assign comfyui-rocm = pkgsRocm.comfyui (modulo handling the infinite recursion)

---

for now I defined the `-cuda` and `-rocm` variants

---

> I think the correct way to.handle this is to introduce pkgsRocm/pkgsCuda similar to the pkgsStatic. The we just assign comfyui-rocm = pkgsRocm.comfyui (modulo handling the infinite recursion)

That wouldn't be too hard, would simplify a LOT of things, and would actually make this PR simpler since we could just put ``comfyui`` under ``pkgs(Cuda|Rocm)``.
That said, I think we may want to ask some of the more long-standing nixpkgs members as to whether we *should* do this or not. Probably need to make a separate issue.

---

We may want to consider something like ``pkgsGPU.(rocm|cuda)``.

---

> That wouldn't be too hard, would simplify a LOT of things, and would actually make this PR simpler since we could just put comfyui under pkgs(Cuda|Rocm).

Anyway, I think it's unreasonable to ask that every package that depends on torch declare two variants.
Why is it so? To make hydra build and cache both version? Aren't both unfree and thus not cached?

---

rocm is free now!

---

But yeah, otherwise I don't see much point in those attributes. Personally, I only use `nix (build|run) -f '<nixpkgs>' --arg config '{ cudaSupport = X; rocmSupport = Y; }' ...`, instead of the `xWithY` attributes

---

Or alternatively, making it easy to override what `torch` is with an overlay. (it's not easy, right know).